webui/ocw/lib/vault.py

```diff
@@ -11,53 +11,42 @@ class Vault:
     auth_json = None
     auth_expire = None
 
-    def __init__(self, url, user, password, certificate_dir):
-        self.url = url
-        self.user = user
-        self.password = password
-        self.certificate_dir = certificate_dir
+    def __init__(self):
         cfg = ConfigFile()
-        if (self.url is None):
-            self.url = cfg.get(['vault', 'url'])
-            self.user = cfg.get(['vault', 'user'])
-            self.password = cfg.get(['vault', 'password'])
-            self.certificate_dir = cfg.get(['vault', 'cert_dir'], '/etc/ssl/certs')
+        self.url = cfg.get(['vault', 'url'])
+        self.user = cfg.get(['vault', 'user'])
+        self.password = cfg.get(['vault', 'password'])
+        self.certificate_dir = cfg.get(['vault', 'cert_dir'], '/etc/ssl/certs')
 
     def __del__(self):
         self.revoke()
 
     def revoke(self):
         if self.auth_json is None:
             return
-        path = '/v1/sys/leases/revoke'
-        self.httpPost(path, {'lease_id': self.auth_json['lease_id']})
+        self.httpPost('/v1/sys/leases/revoke', {'lease_id': self.auth_json['lease_id']}, self.getClientToken)
         self.auth_json = None
 
     def getClientToken(self):
-        login_path = '/v1/auth/userpass/login'
         if self.client_token is None:
-            try:
-                r = requests.post(self.url+login_path+'/'+self.user,
-                                  json={'password': self.password},
-                                  verify=self.certificate_dir)
-                if 'errors' in r.json():
-                    raise ConnectionError(",".join(r.json()['errors']))
-                self.client_token = r.json()['auth']['client_token']
-            except Exception as e:
-                raise ConnectionError('Vault login failed - {}: {}'.format(type(e).__name__, str(e)))
-
+            r = self.httpPost('/v1/auth/userpass/login'+self.user, data={'password': self.password})
+            self.client_token = {'X-Vault-Token': r.json()['auth']['client_token']}
```

Comment:
This will break `httpGet()`. The headers should be build differently in `httpGet()` now.

Reply:
fixed